ambry-named-mysql/src/main/java/com/github/ambry/named/MySqlNamedBlobDb.java

```diff
@@ -628,19 +619,28 @@ private NamedBlobRecord run_get_v2(String accountName, String containerName, Str
   private Page<NamedBlobRecord> run_list_v2(String accountName, String containerName, String blobNamePrefix,
       String pageToken, short accountId, short containerId, Connection connection, Integer maxKeys) throws Exception {
     String query = "";
-    String queryStatement = blobNamePrefix == null ? LIST_ALL_QUERY_V2 : LIST_QUERY_V2;
+    String queryStatement = blobNamePrefix == null ? LIST_ALL_QUERY_V2 : LIST_NAMED_BLOBS_SQL;
     int maxKeysValue = maxKeys == null ? config.listMaxResults : maxKeys;
     try (PreparedStatement statement = connection.prepareStatement(queryStatement)) {
-      statement.setInt(1, accountId);
-      statement.setInt(2, containerId);
       if (blobNamePrefix == null) {
+        // list-all no prefix
+        statement.setInt(1, accountId);
+        statement.setInt(2, containerId);
         statement.setString(3, pageToken);
         statement.setString(4, pageToken);
+        statement.setInt(5, maxKeysValue + 1);
       } else {
+        // list with prefix
+        statement.setInt(1, accountId);
+        statement.setInt(2, containerId);
         statement.setString(3, blobNamePrefix + "%");
         statement.setString(4, pageToken != null ? pageToken : blobNamePrefix);
+        statement.setInt(5, accountId);
+        statement.setInt(6, containerId);
+        statement.setString(7, blobNamePrefix + "%");
+        statement.setString(8, pageToken != null ? pageToken : blobNamePrefix);
+        statement.setInt(9, maxKeysValue + 1);
```

Comment:
Can we add some time like this:
blobName1 blobId1 version1
blobName1 blobId2 version2
so listing we should return blobName1 blobId1 version2 right?
but I think in some case we might return the blobName1 blobId2 version.
This is ok but we should not return the  NamedBlobRecord with blobId info. Instead, we can only return the blobName, cause blobId1 and blobId2 all map to blobName1 and we don't need blobId for listing.

Reply:
This will never happen because we don't combine columns from different rows. The select statement at the end only returns rows from the left side of the join.

Reply:
Yeah, I understand this will never happen for your query, but we want to make sure in the future if other person update this query, things will still work.

Reply:
discussed offline, however i want to add that the next person should be responsible for adding a test if that case is likely to happen.